internal/common/rules/globalruleregistration.go

```diff
@@ -28,55 +28,60 @@ func registerDemoRule() {
 	aux := []float64{1, 1, 2, 0}
 	AuxiliaryVector := mat.NewVecDense(4, aux)
 
-	_, err := RegisterNewRule(name, reqVar, *CoreMatrix, *AuxiliaryVector)
-	if err != nil {
-		panic(err)
+	_, ruleErr := RegisterNewRule(name, reqVar, *CoreMatrix, *AuxiliaryVector, false)
+	if ruleErr != nil {
+		panic(ruleErr.Error())
```

Comment:
might want to actually handle this error now? or is this demo rule never going to be used?

Reply:
Yeah the demo rule is just an example, it will likely be removed in the post-mvp IIGO work